addons/sale/models/sale_order_line.py

```diff
@@ -1179,6 +1187,7 @@ def _check_combo_item_id(self):
     def _onchange_product_id_warning(self):
         if not self.product_id:
             return
+        self._reset_price_unit()
```

Comment:
Unless it's causing issues, I'd prefer if it was in a new dedicated onchange.  It's a bit confusing for the `_onchange_product_id_warning` to handle that as well.